### PR TITLE
restructure repo with fetch/process structure

### PR DESCRIPTION
**Pull request by aaarchmiller**
Changed working directory structure from:

_my_work_R/my_happy_script.R_

to:

_fetch/src/my_happy_script.R_ (moved)
_fetch/out_ (empty directory)
_process/src_ (empty directory)
_process/out_ (empty directory)

### CODE REVIEW

File: 1_fetch/src/functions.R [hunk 1]
```diff
@@ -0,0 +1,15 @@
+fetch <- function(sb_id){
```
Comment (padilla410) [inline 1_fetch/src/functions.R:1]: Good. I also recommend pulling out the `project_output_dir` from the body of the function and making it an argument.

File: 1_fetch/src/functions.R [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  # Define where the output will go
+  project_output_dir <- '1_fetch/out'
+  
+  # Get the data from ScienceBase
+  mendota_file <- file.path(project_output_dir, 'model_RMSEs.csv')
+  sbtools::item_file_download(sb_id, names = 'me_RMSE.csv', destinations = mendota_file, overwrite_file = TRUE)
+  
+  # Return mendota_file
+  #assign(x = "mendota_file", value = mendota_file, envir = .GlobalEnv)
```
Comment (padilla410) [inline 1_fetch/src/functions.R:10]: We're still working through this training, but you will want to clean this piece up before finalizing.

File: 2_process/src/functions.R [hunk 3]
```diff
@@ -0,0 +1,25 @@
+process <- function(input){
+  
+  # Set output directory
+  project_output_dir <- "2_process/out"
```
Comment (padilla410) [inline 2_process/src/functions.R:4]: Similar comment in the `fetch` function: pull `project_output_dir` out of the body and make it an argument. These paths frequently change when working out a new pipeline. it also adds a bit of clarity for someone reviewing your top-level `_targets.R` file.

File: 2_process/src/functions.R [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  
+  # Prepare the data for plotting
+  eval_data <- readr::read_csv(input, col_types = 'iccd') %>%
+    filter(str_detect(exper_id, 'similar_[0-9]+')) %>%
+    mutate(col = case_when(
+      model_type == 'pb' ~ '#1b9e77',
+      model_type == 'dl' ~'#d95f02',
+      model_type == 'pgdl' ~ '#7570b3'
```
Comment (padilla410) [inline 2_process/src/functions.R:12]: I recommend pulling out the color values and making it a vector argument that can be modified as needed.

File: 2_process/src/functions.R [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    ), pch = case_when(
+      model_type == 'pb' ~ 21,
+      model_type == 'dl' ~ 22,
+      model_type == 'pgdl' ~ 23
```
Comment (padilla410) [inline 2_process/src/functions.R:16]: I recommend pulling out the `pch` values and making it a vector argument that can be modified as needed.

File: 3_visualize/src/functions.R [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+                      dl_980mean = filter(input_data, model_type == 'dl', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_980mean = filter(input_data, model_type == 'pb', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_500mean = filter(input_data, model_type == 'dl', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_500mean = filter(input_data, model_type == 'pb', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_100mean = filter(input_data, model_type == 'dl', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_100mean = filter(input_data, model_type == 'pb', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
+                      pgdl_2mean = filter(input_data, model_type == 'pgdl', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_2mean = filter(input_data, model_type == 'pb', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2))
```
Comment (padilla410) [inline 3_visualize/src/functions.R:71]: This is a lot of repeated code. I recommend replacing this with a function.

File: run_all_script.R [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+rm(list = ls())
+
+# Source functions
+source(file = '1_fetch/src/functions.R')
+source(file = '2_process/src/functions.R')
+source(file = '3_visualize/src/functions.R')
+
+# 1 Fetch the data by defining sb_id
+sb_id = '5d925066e4b0c4f70d0d0599'
```
Comment (padilla410) [inline run_all_script.R:16]: Nice 👍

File: run_all_script.R [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+# 2 Process data for plotting and save as .csv
+process(input = rawdata_dir)
+
+processed_data_dir <- '2_process/out/model_summary_results.csv'
+
+# 3 Visualize the results and write the diagnostic log 
+viz(input_dir = processed_data_dir)
+
+whisker(input_dir = processed_data_dir)
```
Comment (padilla410) [inline run_all_script.R:29]: See "overarching recommendations" - I recommend renaming this function to something more descriptive than `whisker`

**Review by padilla410 — CHANGES_REQUESTED** [review]
First, nice job. I was able to successfully run this and produce the expected results.

In addition to the line-by-line comments, I have a few overarching recommendations:
 * Include more descriptive function names; for example in `3_visualize/src/functions.R` the functions `viz` and `whisker` don't communicate much to a collaborator. My first thought when I saw the `whisker` function was "does this have a bar-and-whisker plot?"
 * Add in more arguments to the functions. There are two things that I almost always include in arguments: (1) in and out directories, and (2) style choices (e.g., `pch` values and colors).

File: 1_fetch/src/functions.R (hunk 1 above)
Comment (aaarchmiller) [reply 1_fetch/src/functions.R:1]: Thanks!

File: 1_fetch/src/functions.R (hunk 2 above)
Comment (aaarchmiller) [reply 1_fetch/src/functions.R:10]: Thanks! I meant to do that before pushing, thanks for catching.

**Comment by aaarchmiller** [comment]
Updates as of 2022-06-02 11:52am CT

Resolved all suggestions from Julie, including:

1. renamed functions to be more descriptive
2. made input and output directories into arguments for all functions
3. documented each function file
4. documented the main script file in R-Oxygen for future knitting to html, including titles, organization, headers/footer, etc. 
5. reduced redundancies in the diagnostic log function with for-loop 
6. made new input arguments that included model types, experimental temperatures, colors and pch values
7. forced empty out/ folder directory structure to commit with .empty text files

File: run_all_script.R [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+library(dplyr)
+library(readr)
+library(stringr)
+library(sbtools)
+library(whisker)
+library(devtools)
+
+#' Clear environment (optional)
+#' 
+rm(list = ls())
```
Comment (padilla410) [inline run_all_script.R:19]: These lines can come out. Jenny Bryan from rstudio has a helpful write up on this [here](https://www.tidyverse.org/blog/2017/12/workflow-vs-script/)

File: 2_process/src/process_fcn.R [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+  eval_data <- readr::read_csv(input_data, col_types = 'iccd') %>%
+    filter(str_detect(exper_id, 'similar_[0-9]+')) %>%
+    mutate(col = case_when(
+      model_type == 'pb' ~ col_model_type[1],
+      model_type == 'dl' ~ col_model_type[2],
+      model_type == 'pgdl' ~ col_model_type[3]
+    ), pch = case_when(
+      model_type == 'pb' ~ pch_model_type[1],
+      model_type == 'dl' ~ pch_model_type[2],
+      model_type == 'pgdl' ~ pch_model_type[3]
```
Comment (padilla410) [inline 2_process/src/process_fcn.R:25]: :+1

File: 3_visualize/src/diagnostic_log_fcn.R [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+      # filter data based on model type and exper id
+      filterdata <- input_data[input_data$model_type == model_type[mm] & 
+                                 input_data$n_prof == exper_num[ee],]
+      
+      # calculate mean RMSE and store in list
+      templist <- list(filterdata %>% pull(rmse) %>% mean %>% round(2))
+      names(templist) <- paste0(model_type[mm], "_", exper_num[ee], "mean")
+      render_data <- append(x = render_data, values = templist)
+      }
+  }
```
Comment (padilla410) [inline 3_visualize/src/diagnostic_log_fcn.R:33]: Nice crack at this one. Two comments:
* I would take it one step further and probably pull this thing out and make it a helper function
* appending to an empty table can be a big time sink for large datasets. This pattern works well here, but would probably need a bit more thought if you were to scale it up.

**Review by padilla410 — APPROVED** [review]
I bungled the comment order a bit - see this [comment ](https://github.com/aaarchmiller/ds-pipelines-targets-1/pull/4#issuecomment-1145112526) - approved, but please be sure to check out the comments (specifically, remove `rm(list = ls())`).

Merge when you're ready

**Pull request merged**